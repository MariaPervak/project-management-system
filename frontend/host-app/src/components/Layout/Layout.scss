.layout{
  &__content{
    display: flex;

    .sidebar{
      flex: 0 0 200px;
    }
  }

  &__body{
    background: #f5f7fa;
    flex: 1 1 auto;
    padding: 10px 25px;
  }
}