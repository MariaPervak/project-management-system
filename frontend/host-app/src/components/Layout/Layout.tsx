import './Layout.scss';
import Header from "../Header/Header.tsx";
import Sidebar from "../Sidebar/Sidebar.tsx";


const Layout = ({children}) => {
  return (
    <div className="layout">
      <Header />
      <div className="layout__content">
        <Sidebar />
        {children}
      </div>

    </div>
  );
};

export default Layout;