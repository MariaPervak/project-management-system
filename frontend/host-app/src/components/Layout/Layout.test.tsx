import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Layout from './Layout';

jest.mock('../Header/Header', () => () => <header data-testid="header">Header</header>);
jest.mock('../Sidebar/Sidebar', () => () => <nav data-testid="sidebar">Sidebar</nav>);

describe('Layout component', () => {
    const renderLayout = (children: React.ReactNode = 'Test Content') => {
        return render(<Layout>{children}</Layout>);
    };

    it('renders without crashing', () => {
        renderLayout();
        expect(screen.getByTestId('header')).toBeInTheDocument();
    });

    it('renders all main sections', () => {
        renderLayout();

        expect(screen.getByTestId('header')).toBeInTheDocument();
        expect(screen.getByTestId('sidebar')).toBeInTheDocument();
        expect(screen.getByRole('main')).toBeInTheDocument();
    });

    it('renders children content correctly', () => {
        const testContent = 'Test Child Content';
        renderLayout(testContent);

        expect(screen.getByText(testContent)).toBeInTheDocument();
        expect(screen.getByRole('main')).toHaveTextContent(testContent);
    });

    it('has correct CSS classes', () => {
        renderLayout();

        expect(screen.getByTestId('header')).toBeInTheDocument(); // Проверяем что header есть
        const layoutElement = screen.getByTestId('header').parentElement;
        expect(layoutElement).toHaveClass('layout');

        const contentElement = screen.getByRole('main').parentElement;
        expect(contentElement).toHaveClass('layout__content');

        expect(screen.getByRole('main')).toHaveClass('layout__body');
    });

    it('matches snapshot with children', () => {
        const { asFragment } = renderLayout('Snapshot Content');
        expect(asFragment()).toMatchSnapshot();
    });

    it('matches snapshot without children', () => {
        const { asFragment } = renderLayout();
        expect(asFragment()).toMatchSnapshot();
    });
});