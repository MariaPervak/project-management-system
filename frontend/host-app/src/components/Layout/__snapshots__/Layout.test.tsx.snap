// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component matches snapshot with children 1`] = `
<DocumentFragment>
  <div
    class="layout"
  >
    <header
      data-testid="header"
    >
      Header
    </header>
    <div
      class="layout__content"
    >
      <nav
        data-testid="sidebar"
      >
        Sidebar
      </nav>
      <main
        class="layout__body"
      >
        Snapshot Content
      </main>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Layout component matches snapshot without children 1`] = `
<DocumentFragment>
  <div
    class="layout"
  >
    <header
      data-testid="header"
    >
      Header
    </header>
    <div
      class="layout__content"
    >
      <nav
        data-testid="sidebar"
      >
        Sidebar
      </nav>
      <main
        class="layout__body"
      >
        Test Content
      </main>
    </div>
  </div>
</DocumentFragment>
`;
